warehouse/packaging/services.py

```diff
@@ -369,28 +369,26 @@ def create_service(cls, context, request):
 
 @implementer(IProjectService)
 class ProjectService:
-    def __init__(self, session, remote_addr, metrics=None, ratelimiters=None) -> None:
+    def __init__(self, session, metrics=None, ratelimiters=None) -> None:
         if ratelimiters is None:
             ratelimiters = {}
 
         self.db = session
-        self.remote_addr = remote_addr
         self.ratelimiters = collections.defaultdict(DummyRateLimiter, ratelimiters)
         self._metrics = metrics
 
-    def _check_ratelimits(self, creator):
+    def _check_ratelimits(self, request, creator):
         # First we want to check if a single IP is exceeding our rate limiter.
-        print(self.ratelimiters)
-        if self.remote_addr is not None:
-            if not self.ratelimiters["project.create.ip"].test(self.remote_addr):
+        if request.remote_addr is not None:
+            if not self.ratelimiters["project.create.ip"].test(request.remote_addr):
```

Comment:
The next refactor is likely to replace this with hashed_ip_address values, but that's next time around.

Reply:
we're a ways (couple prs and a big plan) off from the big switch, but yeah.